java/vector/src/main/java/org/apache/arrow/vector/types/Dictionary.java

```diff
@@ -18,23 +18,48 @@
  ******************************************************************************/
 package org.apache.arrow.vector.types;
 
-import org.apache.arrow.vector.ValueVector;
+import org.apache.arrow.vector.FieldVector;
+
+import java.util.Objects;
 
 public class Dictionary {
 
-    private ValueVector dictionary;
-    private boolean ordered;
+  private Long id;
+  private FieldVector dictionary;
+  private boolean ordered;
+
+  public Dictionary(FieldVector dictionary) {
+    this(dictionary, null, false);
+  }
+
+  public Dictionary(FieldVector dictionary, Long id, boolean ordered) {
+    this.id = id;
+    this.dictionary = dictionary;
+    this.ordered = ordered;
+  }
+
+  public Long getId() { return id; }
+
+  public FieldVector getVector() {
+  return dictionary;
```

Comment:
sure, I was trying to match the other files which seem to be 2 space indented. Is 4 correct?